src/domain/hanke/hankeCreateDialog/HankeCreateDialog.tsx

```diff
@@ -0,0 +1,128 @@
+import { Box } from '@chakra-ui/react';
+import {
+  Button,
+  Dialog,
+  TextInput as HDSTextInput,
+  IconCheck,
+  IconCross,
+  IconInfoCircle,
+  Notification,
+} from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useMutation } from 'react-query';
+import { yupResolver } from '@hookform/resolvers/yup';
+import TextInput from '../../../common/components/textInput/TextInput';
+import { newHankeSchema } from '../edit/hankeSchema';
+import useUser from '../../auth/useUser';
+import useLinkPath from '../../../common/hooks/useLinkPath';
+import { ROUTES } from '../../../common/types/route';
+import { createHanke } from '../edit/hankeApi';
+import { NewHankeData } from '../edit/types';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+function HankeCreateDialog({ isOpen, onClose }: Readonly<Props>) {
+  const { t } = useTranslation();
+  const user = useUser();
+  const navigate = useNavigate();
+  const getEditHankePath = useLinkPath(ROUTES.EDIT_HANKE);
+  const formContext = useForm<NewHankeData>({
+    mode: 'onTouched',
+    resolver: yupResolver(newHankeSchema),
+  });
+  const { getValues, trigger, reset: resetForm } = formContext;
+  const { mutate, reset: resetMutation, isLoading, isError } = useMutation(createHanke);
+  const dialogTitle = t('homepage:hanke:title');
+
+  function handleClose() {
+    resetForm();
+    resetMutation();
+    onClose();
+  }
+
+  async function submitForm() {
+    const isFormValid = await trigger(undefined, { shouldFocus: true });
+    if (!isFormValid) {
+      return;
+    }
+
+    mutate(getValues(), {
+      onSuccess({ hankeTunnus }) {
+        handleClose();
+        if (hankeTunnus) {
+          navigate(getEditHankePath({ hankeTunnus }));
+        }
+      },
+    });
+  }
+
+  return (
+    <Dialog
+      id="hanke-create"
+      isOpen={isOpen}
+      aria-labelledby={dialogTitle}
+      variant="primary"
+      close={handleClose}
+      closeButtonLabelText={t('common:ariaLabels:closeButtonLabelText')}
+    >
+      <Dialog.Header
+        id="hanke-create-title"
+        title={dialogTitle}
+        iconLeft={<IconInfoCircle aria-hidden="true" />}
```

Comment:
What is the function of this info icon?

Reply:
It's the icon that is shown in dialog header. It's in Figma: https://www.figma.com/file/CSWoUcI2jslxfOgBYgjKuK/Haitaton-HDS-3.0.0?type=design&node-id=444-9060&mode=design&t=NzN4JxwSEIsWT8yj-0.

Reply:
Okay, so it is just an icon shown in every dialog and it is not suppossed to do anything (like e.g. hovering over or clicking it would show some info label)?

Reply:
Yes, it's just a visual thing.